Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/CallbackJoiner.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+import StitchCore
+import XCTest
+import MongoSwift
+@testable import StitchCoreRemoteMongoDBService
+
+private class SynchronizedDispatchDeque {
```

Comment:
[nit] The classes in this file feel like they should belong in `StitchDarwinCoreTestUtils`. It seems like it would be very useful for all of our async tests. We really abused `XCTestExpectation` in other places and this could really simplify future tests. Is the `@testable` import of the remote mongo service actually necessary for anything here?

Reply:
ah I see that it's for the `ReadWriteLock` Honestly the `ReadWriteLock` should probably go into `StitchCore` since it doesn't actually use any remote mongo service constructs, and then this class can be decoupled from remote mongo tests

Reply:
I agree.

Reply:
I'm going to swap this in the next PR, where I'll need the RWLock actually moved.